src/tracking/lib/tracking.module.ts

```diff
@@ -0,0 +1,40 @@
+import { NgModule, ModuleWithProviders, Injector } from '@angular/core';
+import { TrackingProvider } from './tracking-provider';
+import { MatomoTrackingService } from './matomo-tracking.service';
+import { GATrackingService } from './ga-tracking.service';
+import { AmritTrackingService } from './amrit-tracking.service';
+import { TRACKING_PLATFORM, MATOMO_SITE_ID, MATOMO_URL, TRACKING_PROVIDER } from './tracking.tokens';
+
+export interface TrackingConfig {
+  platform: 'matomo' | 'ga';
+  matomoSiteId?: number;
+  matomoUrl?: string;
+}
+
+@NgModule({})
+export class TrackingModule {
+  static forRoot(config: TrackingConfig): ModuleWithProviders<TrackingModule> {
```

Comment:
setting config like this in the module can lead to issues when we need to alter the individual prop and make it tightly coupled with certain configuration,use environment for configuration related stuff